### ISSUE TITLE
No training scripts and event models as well as inference scripts !

### ISSUE THREAD

**Issue by amitpant7** [event-driven-robotics/hpe-core#114]
Hello, 

I was going through the repo, I checked the examples. The demo and work surprised me. HPE on event data of this level is really astonishing with high accuracy. I wanted to replicate the work. Are those movenet examples of the examples for the move event net. Also, there isn't any training script as well as model and any discription about the data formats. Also it would be great if there was an example to inference your model on a single data instance.

Can you please update this?

Regards 
Amit

**Comment by arrenglover** [comment]
Hi @amitpant7 ! Thanks for your interest. I'll ask @gaurvigoyal to give you some more information!

**Comment by gaurvigoyal** [comment]
Hey Amit! Thanks for your interest. They are in the example folder movenet ([here](https://github.com/event-driven-robotics/hpe-core/tree/main/example/movenet)). There are instructions and a docker for the environment as well. We also provide all of our own models discussed in the ablation study, to support replication.

But I saw that it's a bit stale, since we published it over a year ago. So, I'm checking things, and adding a python-only inference. I'll link the pull request here.

**Comment by amitpant7** [comment]
Thanks for the update. I will check the examples.

**Comment by gaurvigoyal** [comment]
Hey @amitpant7 , we've merged the pull request ([link](https://github.com/event-driven-robotics/hpe-core/pull/116)) with one more option for inference and better instructions. 

We'd like to ensure others can also reach the right folder for this in the future, so I've tried to clarify this in the first Readme.md on [this](https://github.com/event-driven-robotics/hpe-core) page. Would you mind having a look and telling me if it's clear?

**Comment by amitpant7** [comment]
Thanks, the inference is now easy to follow but I couldn't find any scripts to train or refine the networks .

**Comment by gaurvigoyal** [comment]
Hey @amitpant7, 

The code here is only for inference. There is a separate repository with the training code. It's not as clean and as of now the instructions are outdated. If you think access to the training code would benefit your work, we can update that repository and link it to this one.

Let us know!

**Comment by amitpant7** [comment]
Hello @gaurvigoyal 

Thank you for the update! I’d be happy to go through the training script. Additionally, if there are any specific documents outlining the training methods and processes used, I’d greatly appreciate the opportunity to go through those as well.  

Looking forward to your response!

**Comment by DVSexplorer** [comment]
Hey @gaurvigoyal 
I am also looking for relevant training codes, and I would greatly appreciate it if you could link it here.

**Comment by QuangNguyenTang** [comment]
Hi @gaurvigoyal,

Highly appreciated your work on HPE.
I am also looking for the training code to try with other format of events; may I ask if you can provide.
Dirty version is very fine for me.
Thank you.

**Comment by gaurvigoyal** [comment]
Hey all,

Thanks for all your interest in the code. I'm currently adding instructions to the training repository. It would still count as a dirty version, I'd say, but hopefully you can use it to some extent. I'll post the link here as soon as it's in a usable shape.
Ciao!

**Comment by gaurvigoyal** [comment]
Dear @QuangNguyenTang, @DVSexplorer and @amitpant7,

The training code for MoveEnet is available in a separate repository. With a new [merge request](https://github.com/event-driven-robotics/MoveEnet/pull/9), this repository is relatively usable. Here is a link to the repository: [https://github.com/event-driven-robotics/MoveEnet](https://github.com/event-driven-robotics/MoveEnet).
